sycl/include/sycl/ext/intel/pipes.hpp

```diff
@@ -19,6 +19,8 @@ namespace intel {
 
 template <class _name, class _dataT, int32_t _min_capacity = 0> class pipe {
 public:
+  using value_type = _dataT;
+  static constexpr int32_t min_capacity = _min_capacity;
```

Comment:
No worries about lateness. I was on vacation and this is not urgent.
I agree duplicate fields doesn't make sense. What I want is a way for the user of the pipe utility to get the data type and capacity of the pipe without using complicated and difficult to read code stuff. For example, currently I do this:

```c++
template<typename MyPipe>
void foo(/* ... */) {
    using PipeT = decltype(MyPipe::read());
    // ...
}
```

Not all that awful, but not very pretty :/.

Is there anything I can do to solve my problem that doesn't break the ABI? What about static constexpr member functions?

Reply:
I think you are right, probably using member function is more appropriate in this case.
As an example, https://en.cppreference.com/w/cpp/container/array don't expose size as public data member but provides a method. It will also resolve an issue with duplicated data members. And adding a new method is not an ABI-breaking change.

On the other hand, I think my concern about breaking ABI is not justified after all. pipe class is fully defined in the headers only, so pipe objects are not transferred from SYCL RT to user's application and we can't break ABI (between user's application and SYCL RT) in this case.

So, we can go either way: have public data member for size (and remove duplicate data member in this case) or add a new method.

In any case, please add info about these changes to documentation of the pipe: https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/supported/sycl_ext_intel_dataflow_pipes.asciidoc

Reply:
Pipes are only used for FPGA device code and for FPGA we always compile AOT, so I wouldn't worry about ABI breakage here.

Reply:
@MrSidims does that mean you can't have some sort of a library, which you then link with the rest of an application?

Reply:
`std::array` has a `value_type` static member variable. Example: https://godbolt.org/z/rr5dhsY3G

My preference would be to replace `m_Capacity` with a public `min_capacity`, and add a `value_type` static member variable. Thoughts?

Reply:
Yes, I haven't questioned value_type field, it looks perfectly fine. I was referring to the size() method: https://godbolt.org/z/ex9rvYejd

But your current changes look good to me. Could you please mention these new fields in the documentation - https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/supported/sycl_ext_intel_dataflow_pipes.asciidoc? So that users will know that the pipe class has such members.

Reply:
Sorry. My bad on the mix up.
I updated the docs.

Reply:
Thank you!